src/attack.rs

```diff
@@ -0,0 +1,62 @@
+use crate::health::Health;
+use bevy::prelude::*;
+
+// --- Plugin ---
+
+pub struct AttackPlugin;
+
+impl Plugin for AttackPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, attack_target);
+    }
+}
+
+// --- Components ---
+
+#[derive(Component, Debug)]
+pub struct AttackStats {
+    pub damage: i32,
+    pub attack_speed: f32,
+    /// Used to check if the attack target is within striking range.
+    pub attack_range: f32,
+    pub time_till_next_attack: f32,
+}
+
+#[derive(Component)]
+pub struct AttackTarget(pub Entity); // TODO: Make it contain a list of targets.
+                                     // The reason for the above to do is that when a target is destroyed or out of sight
+                                     // new potential targets that are already inside the sensor collider range will not be
+                                     // be used as new targets. We therefore need to store all targets within the vision range.
+                                     // NOTE: Perhaps we could have the list on another component (AttackTargetsInVision)
+                                     // This component would always create a new AttackTarget component with the next target, if none exist.
+
+// --- Systems ---
+
+fn attack_target(
+    mut commands: Commands,
+    mut attacker_query: Query<(Entity, &mut AttackStats, &AttackTarget)>,
```

Comment:
does `AttackStats` change or why is it mutable?

Reply:
On line 43-44 the time till next attack is updated.

This should be made into a Bevy timer (built-in type), but I didn't know at the time.